test/dlc_tests/ec2/test_smdebug.py

```diff
@@ -16,54 +16,58 @@
     if is_tf1(training):
         pytest.skip("Currently skipping for TF1 until the issue is fixed")
-    test_script = SMDEBUG_SCRIPT
-    framework = get_framework_from_image_uri(training)
-    container_test_local_dir = os.path.join("$HOME", "container_tests")
-    ec2_connection.run(f"$(aws ecr get-login --no-include-email --region {region})", hide=True)
-
-    ec2_connection.run(
-        f"nvidia-docker run --name smdebug-gpu -v "
-        f"{container_test_local_dir}:{os.path.join(os.sep, 'test')} -itd {training}",
-        hide=True,
-    )
-
-    test_output = ec2_connection.run(
-        f"nvidia-docker exec --user root smdebug-gpu /bin/bash -c '{test_script} {framework}'",
-        hide=True, warn=True
-    )
-
-    # LOGGER.info(test_output.stdout) # Uncomment this line for a complete log dump
-
-    assert test_output.ok, f"SMDebug tests failed. Output:\n{test_output.stdout}"
+    run_smdebug_test(training, ec2_connection, region, docker_executable="nvidia-docker", container_name="smdebug-gpu")
 
 
+@pytest.mark.flaky(reruns=0)
 @pytest.mark.integration("smdebug")
 @pytest.mark.model("mnist")
 @pytest.mark.parametrize("ec2_instance_type", SMDEBUG_EC2_CPU_INSTANCE_TYPE, indirect=True)
 def test_smdebug_cpu(training, ec2_connection, region, cpu_only, py3_only):
     # TODO: Remove this once test timeout has been debugged (failures especially on m4.16xlarge)
     if is_tf1(training):
         pytest.skip("Currently skipping for TF1 until the issue is fixed")
-    test_script = SMDEBUG_SCRIPT
-    framework = get_framework_from_image_uri(training)
+    run_smdebug_test(training, ec2_connection, region)
+
+
+def run_smdebug_test(
+    image_uri,
+    ec2_connection,
+    region,
+    docker_executable="docker",
+    container_name="smdebug",
+    test_script=SMDEBUG_SCRIPT,
+    logfile="output.log",
+):
+    framework = get_framework_from_image_uri(image_uri)
     container_test_local_dir = os.path.join("$HOME", "container_tests")
     ec2_connection.run(f"$(aws ecr get-login --no-include-email --region {region})", hide=True)
 
     ec2_connection.run(
-        f"docker run --name smdebug-cpu -v {container_test_local_dir}:{os.path.join(os.sep, 'test')} -itd {training}",
+        f"{docker_executable} run --name {container_name} -v "
+        f"{container_test_local_dir}:{os.path.join(os.sep, 'test')} -itd {image_uri}",
         hide=True,
     )
 
-    test_output = ec2_connection.run(
-        f"docker exec --user root smdebug-cpu /bin/bash -c '{test_script} {framework}'",
-        hide=True, warn=True
-    )
+    try:
+        test_output = ec2_connection.run(
+            f"{docker_executable} exec --user root {container_name} "
+            f"/bin/bash -c '{test_script} {framework}' | tee {logfile}",
+            hide=True,
+            warn=True,
+            timeout=3000,
+        )
+    except Exception:
+        debug_output = ec2_connection.run(f"cat {logfile}")
```

Comment:
isn't the logfile inside the docker container?

Reply:
it shouldn't be in this case, basically i'm running the docker exec command and piping it to a file on the host since docker runs on the host